### PR TITLE
 Introducing SiPixelCalSingleMuon ALCARECO

### PR DESCRIPTION
**Pull request by tvami**
Backport of #28535

cc:
@mmusich @tsusa @tocheng @christopheralanwest

### CODE REVIEW

**Comment by christopheralanwest** [comment]
@tvami Is a backport to 10_6_X really needed? [Your presentation at the 2 Dec AlCaDB meeting](https://indico.cern.ch/event/866816/#4-new-alcareco-for-pixel-offli) did not mention that this new ALCARECO was desired for the UL.

**Comment by tvami** [comment]
Yes, please see JIRA ticket: 
https://its.cern.ch/jira/projects/PDMVRERECO/issues/PDMVRERECO-26

**Comment by mmusich** [comment]
This backport is now complete and self-contained.
Can it be tested ?

**Comment by christopheralanwest** [comment]
please test workflow 7.22,7.21,7.4,7.3,138.1

**Comment by chayanit** [comment]
+1

**Comment by tocheng** [comment]
+1

**Comment by silviodonato** [comment]
@tvami  @mmusich could you propagate the fix (#28722) into this PR? Thanks!

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
Have you run a validation of this PR? Could you point here some results?

**Comment by mmusich** [comment]
@silviodonato what kind of results? 
The necessary output files are produced and are filled with the appropriate collections as one can check directly by looking e.g. to the file produced in PR integration tests [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e0b80a/4182/runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017+ALCAFull_2017+NanoFull_2017/) or [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e0b80a/4182/runTheMatrix-results/7.3_CosmicsSPLoose_UP18+CosmicsSPLoose_UP18+DIGICOS_UP18+RECOCOS_UP18+ALCACOS_UP18+HARVESTCOS_UP18/).
For the cosmics case I have also reprocessed an entire run (available [here](https://cernbox.cern.ch/index.php/s/bT7p5XKbNgHyqKM) ).

**Comment by silviodonato** [comment]
merge

**Comment by silviodonato** [comment]
@tvami @mmusich This PR is crashing CMSSW_10_6 2020-01-10-2300
see https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_10_6/2020-01-10-2300?selectedArchs=slc7_amd64_gcc700&selectedFlavors=X&selectedStatus=failed

`The following alcas could not be found ['ALCA,SiPixelCalSingleMuon']
available  ['EcalESAlign', 'LumiPixels', 'PromptCalibProdSiStripGainsAAG', 'SiPixelCalZeroBias', 'PromptCalibProdEcalPedestals', 'TkAlMuonIsolatedPA', 'TkAlUpsilonMuMu', 'EcalCalEtaCalib', 'HcalCalIsolatedBunchSelector', 'HcalCalHBHEMuonFilter', 'MuAlGlobalCosmics', 'MuAlZMuMu', 'TkAlBeamHalo', 'HcalCalIsoTrk', 'PromptCalibProdLumiPCC', 'MuAlGlobalCosmicsInCollisions', 'DtCalibCosmics', 'TkAlCosmicsInCollisions', 'RawPCCProducer', 'MuAlOverlaps', 'TkAlMinBias', 'EcalTestPulsesRaw', 'MuAlBeamHaloOverlaps', 'Hotline', 'TkAlCosmics0T', 'EcalUncalWElectron', 'PromptCalibProdSiPixel', 'PromptCalibProd', 'SiPixelCalSingleMuon', 'SiStripCalMinBiasAAG', 'SiStripCalZeroBias', 'TkAlMuonIsolated', 'PromptCalibProdBeamSpotHP', 'PromptCalibProdSiStrip', 'TkAlCosmics0THLT', 'PromptCalibProdSiPixelAli', 'TkAlUpsilonMuMuPA', 'EcalCalWElectron', 'TkAlLAS', 'TkAlZMuMu', 'AlCaPCCZeroBiasFromRECO', 'EcalTrg', 'TkAlJpsiMuMu', 'RpcCalHLT', 'SiStripCalCosmics', 'AlCaPCCRandom', 'SiStripPCLHistos', 'HcalCalPedestal', 'SiStripCalSmallBiasScan', 'HcalCalIterativePhiSym', 'TkAlCosmics', 'AlCaPCCZeroBias', 'TkAlZMuMuPA', 'DtCalib', 'SiPixelCalCosmics', 'HcalCalDijets', 'PromptCalibProdSiStripGains', 'AlCaPCCRandomFromRECO', 'PromptCalibProdBeamSpotHPLowPU', 'EcalCalPi0Calib', 'HcalCalLowPUHBHEMuonFilter', 'LumiPixelsMinBias', 'HcalCalGammaJet', 'EcalCalZElectron', 'HcalCalIsolatedBunchFilter', 'TkAlCosmicsHLT', 'HcalCalIsoTrkFilter', 'HcalCalNoise', 'HcalCalHOCosmics', 'MuAlCalIsolatedMu', 'SiStripCalMinBias', 'MuAlBeamHalo', 'EcalUncalZElectron', 'HcalCalHO', 'HcalCalMinBias', 'HcalCalHEMuonFilter']
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/nweek-02610/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_10_6_X_2020-01-10-2300/bin/slc7_amd64_gcc700/cmsDriver.py", line 56, in &lt;module&gt;
    run()
  File "/cvmfs/cms-ib.cern.ch/nweek-02610/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_10_6_X_2020-01-10-2300/bin/slc7_amd64_gcc700/cmsDriver.py", line 28, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/nweek-02610/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_10_6_X_2020-01-10-2300/python/Configuration/Applications/ConfigBuilder.py", line 2121, in prepare
    self.addStandardSequences()
  File "/cvmfs/cms-ib.cern.ch/nweek-02610/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_10_6_X_2020-01-10-2300/python/Configuration/Applications/ConfigBuilder.py", line 761, in addStandardSequences
    getattr(self,"prepare_"+stepName)(sequence = ','.join([stepSpec[1],'+'.join(stepSpec[0])]))
  File "/cvmfs/cms-ib.cern.ch/nweek-02610/slc7_amd64_gcc700/cms/cmssw-patch/CMSSW_10_6_X_2020-01-10-2300/python/Configuration/Applications/ConfigBuilder.py", line 1301, in prepare_ALCA
    raise Exception("The following alcas could not be found "+str(alcaList))
Exception: The following alcas could not be found ['ALCA,SiPixelCalSingleMuon']
`

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -2116,15 +2116,15 @@ def gen2018HiMix(fragment,howMuch):
...
 steps['RECODR2_2016reHLT_skimSinglePh']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:SinglePhotonJetPlusHOFilter+EXOMONOPOLE,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2016reHLT']])
 steps['RECODR2_2016reHLT_skimMuOnia']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:BPHSkim,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2016reHLT']])
 
-steps['RECODR2_2017reHLT_skimSingleMu']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@L1TMuon'},steps['RECODR2_2017reHLT']])
+steps['RECODR2_2017reHLT_skimSingleMu']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@L1TMuon'},steps['RECODR2_2017reHLT']])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @tvami, you added by mistake a spurious `ALCA` here that is already there. This crashes the data workflows. Can you fix it?

**Review by mmusich — COMMENTED** [review]
@silviodonato, I identified the problem, but I won't be able to propose a PR until next Wednesday. I hope @tvami can quickly apply the trivial fix.

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 2]
```diff
@@ -2116,15 +2116,15 @@ def gen2018HiMix(fragment,howMuch):
 steps['RECODR2_2017reHLTSiPixelCalZeroBias']=merge([{'--hltProcess':'reHLT','--conditions':'auto:run2_data_relval','-s':'RAW2DIGI,L1Reco,RECO,EI,PAT,ALCA:SiPixelCalZeroBias,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2017']])
 steps['RECODR2_2018reHLTAlCaTkCosmics']=merge([{'--hltProcess':'reHLT','--conditions':'auto:run2_data_relval','-s':'RAW2DIGI,L1Reco,RECO,SKIM:EXONoBPTXSkim,EI,PAT,ALCA:TkAlCosmicsInCollisions,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2018']])
 
-steps['RECODR2_2016reHLT_skimSingleMu']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:ZMu+MuTau,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@L1TMuon'},steps['RECODR2_2016reHLT']])
+steps['RECODR2_2016reHLT_skimSingleMu']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:ZMu+MuTau,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@L1TMuon'},steps['RECODR2_2016reHLT']])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @tvami here too.

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 3]
```diff
@@ -2116,15 +2116,15 @@ def gen2018HiMix(fragment,howMuch):
...
 steps['RECODR2_2016reHLT_skimDoubleEG']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:ZElectron,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2016reHLT']])
-steps['RECODR2_2016reHLT_skimMuonEG']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:TopMuEG,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2016reHLT']])
+steps['RECODR2_2016reHLT_skimMuonEG']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:TopMuEG,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2016reHLT']])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @tvami here too.

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 4]
```diff
@@ -2135,7 +2135,7 @@ def gen2018HiMix(fragment,howMuch):
 steps['RECODR2_2017reHLT_skimCharmonium']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGJPsiSkim+BPHSkim,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM'},steps['RECODR2_2017reHLT']])
 
 steps['RECODR2_2018reHLT']=merge([{'--hltProcess':'reHLT','--conditions':'auto:run2_data_relval'},steps['RECODR2_2018']])
-steps['RECODR2_2018reHLT_skimSingleMu']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+SiStripCalSmallBiasScan+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@L1TMuon'},steps['RECODR2_2018reHLT']])
+steps['RECODR2_2018reHLT_skimSingleMu']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+SiStripCalSmallBiasScan+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@L1TMuon'},steps['RECODR2_2018reHLT']])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @tvami here too.

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 5]
```diff
@@ -2180,7 +2180,7 @@ def gen2018HiMix(fragment,howMuch):
 steps['RECODR2_2017reHLT_skimSinglePh_Prompt']=merge([{'--conditions':'auto:run2_data'},steps['RECODR2_2017reHLT_skimSinglePh']])
 steps['RECODR2_2017reHLT_skimMuOnia_Prompt']=merge([{'--conditions':'auto:run2_data'},steps['RECODR2_2017reHLT_skimMuOnia']])
 steps['RECODR2_2017reHLT_skimCharmonium_Prompt']=merge([{'--conditions':'auto:run2_data'},steps['RECODR2_2017reHLT_skimCharmonium']])
-steps['RECODR2_2017reHLT_skimSingleMu_Prompt_Lumi']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@lumi+@L1TMuon'},steps['RECODR2_2017reHLT_skimSingleMu_Prompt']])
+steps['RECODR2_2017reHLT_skimSingleMu_Prompt_Lumi']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@lumi+@L1TMuon'},steps['RECODR2_2017reHLT_skimSingleMu_Prompt']])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: And here.

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 6]
```diff
@@ -2204,8 +2204,8 @@ def gen2018HiMix(fragment,howMuch):
...
-steps['RECODR2_2018reHLT_skimSingleMu_Offline_Lumi']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@lumi+@L1TMuon'},steps['RECODR2_2018reHLT_skimSingleMu_Offline']])
+steps['RECODR2_2018reHLT_skimSingleMu_Prompt_Lumi']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@lumi+@L1TMuon'},steps['RECODR2_2018reHLT_skimSingleMu_Prompt']])
+steps['RECODR2_2018reHLT_skimSingleMu_Offline_Lumi']=merge([{'-s':'RAW2DIGI,L1Reco,RECO,SKIM:MuonPOGSkim+ZMu+MuTau,EI,PAT,ALCA:ALCA:SiPixelCalSingleMuon+SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@ExtraHLT+@miniAODDQM+@lumi+@L1TMuon'},steps['RECODR2_2018reHLT_skimSingleMu_Offline']])
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: And here.

**Pull request merged**